pinc/ProofreadingToolbox.inc

```diff
@@ -1,375 +1,111 @@
+            new ToolButton('', _('Start block quote'), '/# #/', 'top.surroundSelection("/#\\n","\\n#/");', true),
+            new ToolButton('', _('Thought Break'), '&lt;tb&gt;', 'top.surroundSelection("\\n<tb>\\n","");', true),
+            new ToolButton('', '[Blank Page]', '[Blank Page]', 'top.replaceAllText("[Blank Page]");'),
+        ];
     }
 
-    private function initialize_tool_links()
+    public function draw_tools($round)
     {
-        $this->_tool_links = array();
-        $this->define_tool_link(
-            'greek',
-            '[Greek: ]',
-            '[Greek:',
-            "top.surroundSelection('[Greek: ',']')"
-        );
-        $this->define_tool_link(
-            'sidenote',
-            '[Sidenote: ]',
-            '[Sidenote:',
-            "top.surroundSelection('[Sidenote: ',']')"
-        );
-        $this->define_tool_link(
-            'illustration',
-            '[Illustration: ]',
-            '[Illustration:',
-            "top.surroundSelection('[Illustration: ',']')"
-        );
-        $this->define_tool_link(
-            'note',
-            '[** ]',
-            _('Note'),
-            "top.surroundSelection('[** ',']')"
-        );
-        $this->define_tool_link(
-            'brackets',
-            '[ ]',
-            '[ ]',
-            "top.surroundSelection('[',']')"
-        );
-        $this->define_tool_link(
-            'braces',
-            '{ }',
-            '{ }',
-            "top.surroundSelection('{','}')"
-        );
-        $this->define_tool_link(
-            'footnote',
-            '[Footnote #: ]',
-            '[Footnote #:',
-            "top.surroundSelection('[Footnote #: ',']')"
-        );
-        $this->define_tool_link(
-            'slash_star',
-            '/* */',
-            _('Start poetry/table'),
-            "top.surroundSelection('/*\\n','\\n*/')"
-        );
-        $this->define_tool_link(
-            'slash_number',
-            '/# #/',
-            _('Start block quote'),
-            "top.surroundSelection('/#\\n','\\n#/')"
-        );
-        $this->define_tool_link(
-            'thought_break',
-            '<tb>',
-            _('Thought Break'),
-            "top.surroundSelection('\\n<tb>\\n','')"
-        );
-        $this->define_tool_link(
-            'blank_page',
-            '[Blank Page]',
-            '[Blank Page]',
-            "top.replaceAllText('[Blank Page]')"
-        );
-    }
+        global $code_url;
+        $formatting = is_formatting_round($round);
 
-    private function array_extract( $source, $keys )
-    {
-        if ( $keys == 'ALL' )
+        echo "<div id='tools'>";
+        foreach ($this->tool_buttons as $tool_button)
         {
-            return $source;
+            $tool_button->draw($formatting);
         }
 
-        $result = array();
-        foreach ( $keys as $key )
+        draw_popup_link(_('Search/Replace'), 'srchrep.php', 'dp_searchrepl', 300, 250);
```

Comment:
Consider putting the tool links on a line by themselves.

Reply:
I also think we should retain a "Pop-up tools:" prefix to the line too.

Reply:
Done this.